src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/ResourceScopeImpl.java

```diff
@@ -124,6 +124,27 @@ public static ResourceScopeImpl createShared(Cleaner cleaner) {
         return new SharedScope(cleaner);
     }
 
+    @Override
+    public void release(Handle handle) {
+        try {
+            Objects.requireNonNull(handle);
+            if (!isImplicit() && handle.scope() != this) {
```

Comment:
Is this check correct, since an implicit scope could accept a handle from a non-implicit scope?

Reply:
yeah - something is off here